app/src/main/java/org/sopt/and/component/TextFieldCustom.kt

```diff
@@ -0,0 +1,135 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Info
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusDirection
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.sopt.and.R
+
+@Composable
+fun TextField_Custom(
+    text: String,
+    textfield: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String,
+    showed: Boolean,
+    keyboardOptions: KeyboardOptions,
+    trailingIcon: @Composable (() -> Unit)? = null
+) {
+    val showPassword = if (!showed) {
+        PasswordVisualTransformation()
+    } else {
+        VisualTransformation.None
+    }
+    val focusManager = LocalFocusManager.current
+    Column(modifier = Modifier) {
+        TextField(
+            value = textfield,
+            onValueChange = onValueChange,
+            trailingIcon = trailingIcon,
+            placeholder = { Text(placeholder) },
+            visualTransformation = showPassword, singleLine = true,
+            keyboardOptions = keyboardOptions,
+            keyboardActions = KeyboardActions(
+                onDone = {
+                    focusManager.clearFocus(true)
+                },
+                onNext = {
+                    focusManager.moveFocus(FocusDirection.Down)
+                }),
+            modifier = Modifier
+                .fillMaxWidth(),
+            colors = TextFieldDefaults.colors(
+                unfocusedContainerColor = colorResource(R.color.textfield_gray2f)
+            ),
+        )
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            Icon(
+                imageVector = Icons.Default.Info,
+                contentDescription = Icons.Default.Info.name
+            )
+            Text(
+                text = text,
+                fontSize = 12.sp,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(7.dp), color = colorResource(R.color.gray_a3)
+            )
+        }
+    }
+}
+
+@Composable
+fun TextField_Custom2(
+    textfield: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String,
+    showed: Boolean,
+    keyboardOptions: KeyboardOptions,
+    trailingIcon: @Composable (() -> Unit)? = null
```

Comment:
오

```suggestion
    trailingIcon: @Composable (() -> Unit)? = null
```

와

```suggestion
    trailingIcon: @Composable (() -> Unit) = {}
```

의 의미론적 차이를 알고 계실까요?

Reply:
nullable 같은 경우에 함수 호출시 인자가 전달 되지 않으면 trailingIcon 자체가 생성 되지않는 것으로 알고있습니다.
만약 default parameter를 이용한다면 인자가 전달 되지않더라도 빈함수가 rendering 되고 빈 composable 함수가 항상 존재합니다!
여기서는 {}가 더 좋을것 같네요